--- obs-studio-31.0.0/plugins/CMakeLists.txt.orig	2024-12-07 00:21:55.000000000 +0100
+++ obs-studio-31.0.0/plugins/CMakeLists.txt	2024-12-26 12:09:59.606665425 +0100
@@ -9,27 +9,6 @@
 
 set_property(GLOBAL APPEND PROPERTY OBS_FEATURES_ENABLED "Plugin Support")
 
-macro(check_obs_browser)
-  if((OS_WINDOWS AND CMAKE_VS_PLATFORM_NAME MATCHES "(Win32|x64)") OR OS_MACOS OR OS_LINUX)
-    if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt")
-      message(FATAL_ERROR "Required submodule 'obs-browser' not available.")
-    else()
-      add_subdirectory(obs-browser)
-    endif()
-  else()
-    add_custom_target(obs-browser)
-    target_disable(obs-browser)
-  endif()
-endmacro()
-
-macro(check_obs_websocket)
-  if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket/CMakeLists.txt")
-    message(FATAL_ERROR "Required submodule 'obs-websocket' not available.")
-  else()
-    add_subdirectory(obs-websocket)
-  endif()
-endmacro()
-
 # Add plugins in alphabetical order to retain order in IDE projects
 add_obs_plugin(
     aja
@@ -56,8 +35,6 @@
 add_obs_plugin(mac-virtualcam PLATFORMS MACOS)
 add_obs_plugin(nv-filters PLATFORMS WINDOWS)
 
-check_obs_browser()
-
 add_obs_plugin(obs-ffmpeg)
 add_obs_plugin(obs-filters)
 add_obs_plugin(obs-libfdk)
@@ -77,8 +54,6 @@
 )
 add_obs_plugin(obs-webrtc)
 
-check_obs_websocket()
-
 add_obs_plugin(obs-x264)
 add_obs_plugin(oss-audio PLATFORMS FREEBSD OPENBSD)
 add_obs_plugin(rtmp-services)
