diff -dur obs-studio-19.0.3.orig/libobs/CMakeLists.txt obs-studio-19.0.3/libobs/CMakeLists.txt
--- obs-studio-19.0.3.orig/libobs/CMakeLists.txt	2017-06-22 09:58:45.000000000 +0200
+++ obs-studio-19.0.3/libobs/CMakeLists.txt	2017-07-31 14:19:57.000000000 +0200
@@ -425,6 +425,9 @@
 		${OBS_JANSSON_IMPORT}
 		${FFMPEG_LIBRARIES}
 		${ZLIB_LIBRARIES}
+		${PKG_X11_XCB_LIBRARIES}
+		-ldl
+		-lm
 	PUBLIC
 		${THREADS_LIBRARIES})
 
