diff -urN obs-studio-21.0.1.orig/UI/frontend-plugins/frontend-tools/scripts.cpp obs-studio-21.0.1/UI/frontend-plugins/frontend-tools/scripts.cpp
--- obs-studio-21.0.1.orig/UI/frontend-plugins/frontend-tools/scripts.cpp	2018-01-23 01:44:22.000000000 +0100
+++ obs-studio-21.0.1/UI/frontend-plugins/frontend-tools/scripts.cpp	2019-01-14 10:58:20.471181467 +0100
@@ -2,6 +2,7 @@
 #include "frontend-tools-config.h"
 #include "../../properties-view.hpp"
 
+#include <QAction>
 #include <QFileDialog>
 #include <QPlainTextEdit>
 #include <QHBoxLayout>
