--- obs-studio-32.1.2/libobs-opengl/CMakeLists.txt.orig	2026-04-21 21:00:34.000000000 +0200
+++ obs-studio-32.1.2/libobs-opengl/CMakeLists.txt	2026-06-24 21:32:11.294384433 +0200
@@ -58,6 +58,7 @@ target_link_libraries(
     $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:Libdrm::Libdrm>
     $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:XCB::XCB>
     $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:X11::XCB>
+    $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:X11::X11>
     $<$<AND:$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>,$<BOOL:${ENABLE_WAYLAND}>>:OpenGL::EGL>
     $<$<AND:$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>,$<BOOL:${ENABLE_WAYLAND}>>:Wayland::Egl>
 )
--- obs-studio-32.1.2/libobs/CMakeLists.txt.orig	2026-04-21 21:00:34.000000000 +0200
+++ obs-studio-32.1.2/libobs/CMakeLists.txt	2026-06-24 21:34:35.738063612 +0200
@@ -249,6 +249,7 @@ target_link_libraries(
     jansson::jansson
     Uthash::Uthash
     ZLIB::ZLIB
+    glib-2.0
   PUBLIC SIMDe::SIMDe Threads::Threads
 )
 
