--- obs-studio-31.0.0/libobs-opengl/CMakeLists.txt~	2024-12-07 00:21:55.000000000 +0100
+++ obs-studio-31.0.0/libobs-opengl/CMakeLists.txt	2024-12-26 12:44:37.833332079 +0100
@@ -56,6 +56,7 @@
     "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,IOSurface.framework>>"
     $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:xcb::xcb>
     $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:X11::x11-xcb>
+    $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:X11::X11>
     $<$<AND:$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>,$<BOOL:${ENABLE_WAYLAND}>>:OpenGL::EGL>
     $<$<AND:$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>,$<BOOL:${ENABLE_WAYLAND}>>:Wayland::EGL>
 )
--- obs-studio-31.0.0/libobs/CMakeLists.txt~	2024-12-07 00:21:55.000000000 +0100
+++ obs-studio-31.0.0/libobs/CMakeLists.txt	2024-12-26 12:47:01.733332089 +0100
@@ -7,6 +7,10 @@
   return()
 endif()
 
+if(OS_LINUX OR OS_FREEBSD OR OS_OPENBSD)
+  find_package(X11 REQUIRED)
+endif()
+
 find_package(Threads REQUIRED)
 
 find_package(FFmpeg 6.1 REQUIRED avformat avutil swscale swresample OPTIONAL_COMPONENTS avcodec)
@@ -260,6 +264,7 @@
     jansson::jansson
     Uthash::Uthash
     ZLIB::ZLIB
+    $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:X11::X11>
   PUBLIC Threads::Threads
 )
 
