--- obs-studio-32.1.2/plugins/linux-v4l2/v4l2-input.c.orig	2026-04-21 21:00:34.000000000 +0200
+++ obs-studio-32.1.2/plugins/linux-v4l2/v4l2-input.c	2026-06-24 20:23:54.697993327 +0200
@@ -243,8 +243,8 @@ static void *v4l2_thread(void *vptr)
 			break;
 		}
 
-		blog(LOG_DEBUG, "%s: ts: %06ld buf id #%d, flags 0x%08X, seq #%d, len %d, used %d", data->device_id,
-		     buf.timestamp.tv_usec, buf.index, buf.flags, buf.sequence, buf.length, buf.bytesused);
+		blog(LOG_DEBUG, "%s: ts: %06lld buf id #%d, flags 0x%08X, seq #%d, len %d, used %d", data->device_id,
+		     (long long)buf.timestamp.tv_usec, buf.index, buf.flags, buf.sequence, buf.length, buf.bytesused);
 
 		if (buf.flags & V4L2_BUF_FLAG_ERROR) {
 			blog(LOG_DEBUG, "skipping decoding of buffer with recoverable error-flag set");
--- obs-studio-31.0.0/plugins/obs-qsv11/common_utils_linux.cpp.orig	2024-12-07 00:21:55.000000000 +0100
+++ obs-studio-31.0.0/plugins/obs-qsv11/common_utils_linux.cpp	2024-12-27 00:01:45.876665615 +0100
@@ -509,7 +509,11 @@
 	obs_enter_graphics();
 	uint32_t gs_count = gs_get_adapter_count();
 	if (*adapter_count < gs_count) {
+#if defined(__x86_64__) && defined(__ILP32__)
+		blog(LOG_WARNING, "Too many video adapters: %d < %d", *adapter_count, gs_count);
+#else
 		blog(LOG_WARNING, "Too many video adapters: %ld < %d", *adapter_count, gs_count);
+#endif
 		obs_leave_graphics();
 		return;
 	}
--- obs-studio-31.0.0/shared/bpm/bpm.c.org	2024-12-27 00:27:44.626665761 +0100
+++ obs-studio-31.0.0/shared/bpm/bpm.c	2024-12-27 00:13:58.196665720 +0100
@@ -9,7 +9,11 @@
 	 */
 	memset(&m_time->rfc3339_str, 0, sizeof(m_time->rfc3339_str));
 	strftime(m_time->rfc3339_str, sizeof(m_time->rfc3339_str), "%Y-%m-%dT%T", gmtime(&m_time->tspec.tv_sec));
+#if defined(__x86_64__) && defined(__ILP32__)
+	sprintf(m_time->rfc3339_str + strlen(m_time->rfc3339_str), ".%03lldZ", m_time->tspec.tv_nsec / 1000000);
+#else
 	sprintf(m_time->rfc3339_str + strlen(m_time->rfc3339_str), ".%03ldZ", m_time->tspec.tv_nsec / 1000000);
+#endif
 	m_time->valid = true;
 }
 
