--- obs-studio-29.1.3/plugins/linux-v4l2/v4l2-input.c~	2023-06-19 00:05:22.000000000 +0200
+++ obs-studio-29.1.3/plugins/linux-v4l2/v4l2-input.c	2023-11-05 00:32:37.108931868 +0100
@@ -259,7 +259,11 @@
 			break;
 		}
 
+#if defined(__x86_64__) && defined(__ILP32__)
+		blog(LOG_DEBUG, "%s: ts: %0al6ld buf id #%d, flags 0x%08X, seq #%d, len %d, used %d", data->device_id,
+#else
 		blog(LOG_DEBUG, "%s: ts: %06ld buf id #%d, flags 0x%08X, seq #%d, len %d, used %d", data->device_id,
+#endif
 		     buf.timestamp.tv_usec, buf.index, buf.flags, buf.sequence, buf.length, buf.bytesused);
 
 		out.timestamp = timeval2ns(buf.timestamp);
